src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -1194,6 +1194,7 @@ trait Types
       else super.safeToString
     override def narrow: Type = this
     override def kind = "ThisType"
+    override def boundSyms = if (sym.isInstanceOf[StubSymbol]) emptySymbolSet else super.boundSyms
```

Comment:
> While looking at `boundSyms`, do you know why `ExistentialType.boundSyms` doesn't `++ underlying.boundSyms`?

Looks like it is called repeatedly as the [substituition `TypeMap`](https://github.com/scala/scala/blob/1eb331b4196492f8f8084a2c102272c3a9c0cd2d/src/reflect/scala/reflect/internal/tpe/TypeMaps.scala#L696-L759) recurses, so its doing the right thing by being shallow. 

Come to think of it, I suspect we could actually just return null (in `SimpleTypeProxy.boundSyms`) and have the same results as we do today.